lib/urlencode.js

```diff
@@ -33,7 +33,7 @@ function encode(str, charset) {
   var buf = iconv.encode(str, charset);
   var encodeStr = '';
   for (var i = 0; i < buf.length; i++) {
-    encodeStr += '%' + buf[i].toString('16');
+    encodeStr += '%' + buf.toString('hex', i, i + 1);
```

Comment:
for '\n', "buf[i].toString('16')" get "a", but "buf.toString('hex', i, i + 1)" get "0a".